ros2_serial_example/CMakeLists.txt

```diff
@@ -99,22 +100,43 @@ target_link_libraries(bridge_gen
   fastcdr
 )
 
-add_executable(ros2_to_serial_bridge
+add_executable(ros2_to_serial_bridge_node
   src/ros2_to_serial_bridge.cpp
   src/ros2_to_serial_bridge_main.cpp
   src/uart_transporter.cpp
   src/udp_transporter.cpp
 )
-ament_target_dependencies(ros2_to_serial_bridge
+ament_target_dependencies(ros2_to_serial_bridge_node
   rclcpp
   ros2_serial_msgs
+  rclcpp_components
+)
+target_link_libraries(ros2_to_serial_bridge_node
+  bridge_gen
+  fastcdr
+  ring_buffer
+  transporter
+)
+
+add_library(ros2_to_serial_bridge SHARED
+  src/ros2_to_serial_bridge.cpp
+  src/uart_transporter.cpp
+  src/udp_transporter.cpp
 )
+target_compile_definitions(ros2_to_serial_bridge
+  PRIVATE "COMPOSITION_BUILDING_DLL")
+ament_target_dependencies(ros2_to_serial_bridge
+  "rclcpp"
+  "rclcpp_components"
+  "ros2_serial_msgs")
 target_link_libraries(ros2_to_serial_bridge
   bridge_gen
   fastcdr
   ring_buffer
   transporter
 )
+rclcpp_components_register_nodes(ros2_to_serial_bridge "ros2_to_serial_bridge::ROS2ToSerialBridge")
```

Comment:
This isn't super-well documented at the moment, but if you want to build a component and a standalone executable, you can use

```
rclcpp_components_register_node(
  ros2_to_serial_bridge
  PLUGIN "ros2_to_serial_bridge::ROS2ToSerialBridge"
  EXECUTABLE ros2_to_serial_bridge_node
)```